src/include/btmem.h

```diff
@@ -339,9 +339,10 @@ struct __wt_page_modify {
                     WT_INSERT *ins; /* Insert list reference */
                     WT_ROW *ripcip; /* Original on-page reference */
                     WT_UPDATE *onpage_upd;
+                    bool restore; /* Whether to restore this saved update chain */
                 } * supd;
                 uint32_t supd_entries;
-                bool supd_restore;
+                bool supd_restore; /* Whether to restore saved update chains to this page */
```

Comment:
The relationship between `WT_SAVE_UPD::restore`, `WT_MULTI::supd_restore` and `WT_RECONCILE::cache_write_restore`. Seems complex and fragile. I see why the different layers are used, but I'm nervous that the scheme is going to be error prone and difficult to debug. I already had to think hard about cases where you switched from cache_write_restore and supd_restore, and I've been thinking about how this works recently. Making the right choice in the future will be hard.
I'm not going to hold up this important change, especially since I don't have a better proposal, but please create a two new tickets titled:
* Document how reconciliation tracks, frees and restores updates.
* Improve how we track which updates are restored during reconciliation.
* Improve how we track

Reply:
Created WT-5941 and WT-5942